tests/console/docker.pm

```diff
@@ -50,47 +50,58 @@ sub run {
-    validate_script_output("docker run --name test_2 alpine:$alpine_image_version /bin/echo Hello world",              sub { m/Hello world/ });
-    validate_script_output("docker ps -a",                                                                             sub { m/test_1/ });
-    validate_script_output("docker ps -a",                                                                             sub { m/test_2/ });
-    validate_script_output("docker run --name test_ephemeral --rm alpine:$alpine_image_version /bin/echo Hello world", sub { m/Hello world/ });
+    assert_script_run('docker container run --name test_1 hello-world | grep "Hello from Docker\!"');
+    assert_script_run(qq#docker container run --name test_2 alpine:$alpine_image_version /bin/echo Hello world | grep "Hello world"#);
+    assert_script_run(qq#docker container run --name test_ephemeral --rm alpine:$alpine_image_version /bin/echo Hello world | grep "Hello world"#);
+    my $cmd_docker_container_ls = 'docker container ls -a';
+    my $output_containers       = script_output($cmd_docker_container_ls);
+    die('error: missing container test_1')            if ($output_containers =~ m/test_1/);
+    die('error: missing container test_2')            if ($output_containers =~ m/test_2/);
+    die('error: ephemeral container was not removed') if ($output_containers =~ m/test_ephemeral/);
 
     # containers can be run on background
-    my ($container_id) = script_output("docker run -d -t -i alpine:$alpine_image_version /bin/sh") =~ /(.+)/;
-    script_run("docker inspect --format='{{.State.Running}}' ${container_id}");
-    validate_script_output("docker inspect --format='{{.State.Running}}' ${container_id}", sub { m/true/ });
+    my ($container_id) = script_output("docker container run -d alpine:$alpine_image_version tail -f /dev/null") =~ /(.+)/;
+    assert_script_run("docker container inspect --format='{{.State.Running}}' ${container_id} | grep true");
 
     # containers can be stopped
-    assert_script_run("docker stop ${container_id}");
-
-    # check that alpine container should not be running anymore
-    validate_script_output("docker inspect --format='{{.State.Running}}' ${container_id}", sub { m/false/ });
+    assert_script_run("docker container stop ${container_id}");
+    assert_script_run("docker container inspect --format='{{.State.Running}}' ${container_id} | grep false");
 
     # network is working inside of the containers
-    script_run("docker run --rm alpine:$alpine_image_version wget http://google.com && echo 'container_network_works' > /dev/$serialdev", 0);
-    die("network does not work inside of the container") unless wait_serial("container_network_works", 200);
+    assert_script_run(qq#docker container run --rm alpine:$alpine_image_version wget http://google.com 2>&1 | grep "index.html\\s*100%"#);
```

Comment:
I consider curl more lightweight but wget is also ok. I recommend to use `wget -q …` though and just check for the exit code being 0

Reply:
true `curl` is preferred for such checks, the problem is that `curl` is not installed by default into the `alpine` image.

```bash
docker container run -it --rm alpine /bin/sh
/ # curl
/bin/sh: curl: not found
```

You have to either install it and then use it:
```bash
/ # apk --no-cache add curl  
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
(1/3) Installing ca-certificates (20161130-r2)
(2/3) Installing libcurl (7.58.0-r0)
(3/3) Installing curl (7.58.0-r0)
Executing busybox-1.26.2-r5.trigger
Executing ca-certificates-20161130-r2.trigger
OK: 5 MiB in 14 packages
/ # curl
curl: try 'curl --help' or 'curl --manual' for more information
```

or try another image with `curl` already installed. Personally guys I would prefer to replace all those alpine and busybox images and eat our own food (opensuse image).

Reply:
@drpaneas, that's part of the plan, but for a later PR ;)